flink-runtime/src/main/java/org/apache/flink/runtime/rest/util/RestMapperUtils.java

```diff
@@ -33,8 +33,7 @@
 		objectMapper.enable(
 			DeserializationFeature.FAIL_ON_IGNORED_PROPERTIES,
 			DeserializationFeature.FAIL_ON_NULL_FOR_PRIMITIVES,
-			DeserializationFeature.FAIL_ON_READING_DUP_TREE_KEY,
-			DeserializationFeature.FAIL_ON_MISSING_CREATOR_PROPERTIES);
```

Comment:
Removing this also means we have to add explicit null checks to all existing Request-/ResponseBody classes.

Reply:
Do note that we must be on the look-out for requests that use primitive fields, as jackson will default them to 0 if they are missing, which will cause misleading error messages.

Reply:
This is a valid objection which I share. I'll remove this change and set the map of `TaskCheckpointStatistics` to an empty map in case that we want to leave the details out.